HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/ScreenAPICoordinator.swift

```diff
@@ -81,37 +79,99 @@ final class ScreenAPICoordinator: NSObject, Coordinator, GiniHealthTrackingDeleg
+    }
+
+    private func checkIfDocumentIsPayable(for docId: String, using healthSdk: GiniHealth, completion: @escaping (Bool) -> Void) {
+        healthSdk.checkIfDocumentIsPayable(docId: docId) { resultPayable in
+            switch resultPayable {
+            case .success(let payable):
+                completion(payable)
+            case .failure(let error):
+                GiniUtilites.Log("Check if document is payable failed with: \(String(describing: error))", event: .error)
+                completion(false)
+            }
+        }
+    }
+
+    private func fetchDocumentDataForReview(docId: String, using healthSdk: GiniHealth) {
+        healthSdk.fetchDataForReview(documentId: docId) { [weak self] resultReview in
+            switch resultReview {
+            case .success(let data):
+                self?.fetchExtractions(for: data.document, healthSdk: healthSdk, docId: docId)
+            case .failure(let error):
+                GiniUtilites.Log("Document data fetching failed: \(String(describing: error))", event: .error)
+            }
+        }
+    }
+
+    private func fetchExtractions(for document: GiniHealthSDK.Document, healthSdk: GiniHealth, docId: String) {
+        healthSdk.documentService.extractions(for: document, cancellationToken: CancellationToken()) { [weak self] result in
+            switch result {
+            case .success(let extractionResult):
+                GiniUtilites.Log("Successfully fetched extractions for id: \(docId)", event: .success)
+                self?.handleSuccessfulExtractions(extractionResult, for: docId)
+            case .failure(let error):
+                GiniUtilites.Log("Obtaining extractions from document with id \(docId) failed with error: \(String(describing: error))", event: .error)
+            }
+        }
+    }
+
+    private func handleSuccessfulExtractions(_ extractionResult: GiniHealthSDK.ExtractionResult, for docId: String) {
+        let invoice = DocumentWithExtractions(documentId: docId, extractionResult: extractionResult)
+        self.hardcodedInvoicesController.appendInvoiceWithExtractions(invoice: invoice)
+
+        DispatchQueue.main.async {
+            self.rootViewController.dismiss(animated: true) {
+                self.delegate?.presentInvoicesList(invoices: [invoice])
+            }
+        }
+    }
+
+    private func presentErrorForNonPayableDocument() {
+        DispatchQueue.main.async {
+            self.rootViewController.dismiss(animated: true) {
+                self.delegate?.presentError(title: NSLocalizedString("gini.health.example.error.invoice.not.payable", comment: ""), message: "")
+            }
+        }
+    }
+
+    private func presentErrorForMultipleInvoicesInDocument() {
+        DispatchQueue.main.async {
+            self.rootViewController.dismiss(animated: true) {
+                self.delegate?.presentError(title: NSLocalizedString("gini.health.example.error.contains.multiple.invoices", comment: ""), message: "")
```

Comment:
[request] Can we have a separate property for the title to avoid having swiftlint warnings because the line can become longer?